namenode/src/main/java/org/apache/crail/namenode/BlockStore.java

```diff
@@ -107,6 +107,85 @@ short prepareDataNodeForRemoval(DataNodeInfo dn) throws IOException {
 		return RpcErrors.ERR_DATANODE_NOT_REGISTERED;
 	}
 
+	public double getStorageUsage() throws Exception {
+		long total = 0;
+		long free = 0;
+		for(StorageClass storageClass : storageClasses) {
+			total += storageClass.getTotalCapacity();
+			free += storageClass.getFreeCapacity();
+		}
+
+		// if there is no available capacity (i.e. total number of available blocks is 0),
+		// return 1.0 which tells that all storage is used
+		if(total != 0) {
+			double available = (double) free / (double) total;
+			return 1.0 - available;
+		} else {
+			return 1.0;
+		}
+
+	}
+
+	public int getBlockUsage() throws Exception {
```

Comment:
Should return `long` as `getTotalCapacity` and `getFreeCapactiy` return long. Otherwise this could overflow.
Also Naming: Maybe `getNumberOfBlocksUsed`